build_scripts/docker_image_creator.py

```diff
@@ -0,0 +1,66 @@
+""" Script to create Sagemaker TensorFlow Docker images
+
+    Usage:
+        python docker_image_creator.py optimized_binary_link gpu|cpu tensorflow_version python_version
+"""
+import argparse
+import glob
+import subprocess
+import sys
+
+## GLOBALS
+DOCKER = 'docker' # Stores which type of DOCKER to use, nvidia-docker or docker
+PATH_TO_SCRIPT = ''  # Stores route the user takes to build scripts directory
+
+def create_docker_image(optbin_link, processor, framework_version, python_version):
+    """ Function builds a docker image with the TF optimized binary
+
+    Args:
+        optbin_link (str): link to where the optimized binary is
+        processor (str): gpu or cpu
+        framework_version (str): tensorflow version i.e 1.6.0
+        python_version (str): (i.e. 3.6.5 or 2.7.4)
+    """
+    # 1.) Initialize commonly used variables
+    py_v = 'py{}'.format(python_version.split('.')[0]) # i.e. py2
+
+    # 2.) Get optimized binary - and put in final docker image repo
+    print('Getting optimized binary...')
+    optbin_filename = 'tensorflow-{}-cp27-cp27mu-manylinux1_x86_64.whl'.format(framework_version)
+    with open('{}/../docker/{}/final/{}/{}'.format(PATH_TO_SCRIPT, framework_version, py_v, optbin_filename), 'wb') as optbin_file:
+        subprocess.call(['curl', optbin_link], stdout=optbin_file)
+
+    # 3.) Build base image
```

Comment:
If the base already exists, docker won't rebuild it. It will check that the exact image already exists and continue. So, if you already have the base image in the directory, it will take 10 seconds as opposed to 20 minutes to go through this step.